### ISSUE TITLE
Failure to handle data with column names containing '.'

### ISSUE THREAD

**Issue by tomtor** [open-metadata/OpenMetadata#981]
When running metadata ingest for a PostgreSQL source I get:

```
[2021-10-30 10:20:25,853] ERROR    {metadata.ingestion.source.sql_source:187} - Failed to generate sample data for tunneldeel - 2 validation errors for TableData
columns -> 3 -> __root__
  string does not match regex "^[^.]*$" (type=value_error.str.regex; pattern=^[^.]*$)
columns -> 4 -> __root__
  string does not match regex "^[^.]*$" (type=value_error.str.regex; pattern=^[^.]*$)
```

Column names may not contain a '.' character? Colum names for column 3 and 4 indeed have a `.`

```
 identificatie.namespace  identificatie.lokaalid
```

**Comment by tomtor** [comment]
These column names are currently dropped from the metadata (best would be to remap eg `.` to `_`), but to be consistent the sample data should/could still be generated.

**Comment by harshach** [comment]
@tomtor Thanks for filing this. let us know if you are interested in taking this up. Thanks

**Comment by sureshms** [comment]
> These column names are currently dropped from the metadata (best would be to remap eg `.` to `_`), but to be consistent the sample data should/could still be generated.
> 
> EDIT: A strategy could be to rename `.` in (column) names to `_DOT_` and remap `_DOT_` to `.` when querying.

Great idea. Let's do it 👍.

**Comment by tomtor** [comment]
> @tomtor Thanks for filing this. let us know if you are interested in taking this up. Thanks

Sure after https://github.com/open-metadata/OpenMetadata/pull/980 is done. I have to get more familiar with the code base.

**Comment by harshach** [comment]
I don't see any reason to map to special chars here. We can preserve the dot-notation it won't break anything on the backend or UI side. Looks like a bug to me on the ingestion parsing side

**Comment by sureshms** [comment]
It will break EntityUtil#getLocalColumnName method. The only way to handled dot in the names, is allow it only in column name and change this method to handle such column names. If dot is allowed in service/database/table names, then there is no way to generically solve it other than introducing _DOT_ proposed by @tomtor.

**Comment by harshach** [comment]
Ya forgot about FQN. makes sense with DOT option

**Comment by tomtor** [comment]
I now have working code with the DOT mapping.

I can merge it with #980 or is a separate PR prefered?

**Comment by harshach** [comment]
@tomtor i merged the #980 . Can you open a nee PR. Thansk
